### PR TITLE
Tensor operators

### PR DESCRIPTION
**Pull request by alewis**
Adds *, -, +, /, and @ operators for tn.Tensor.

### CODE REVIEW

**Comment by mganahl** [comment]
@alewis I'll merge once the conflicts are fixed

File: tensornetwork/linalg/operations.py [hunk 1]
```diff
@@ -343,4 +343,4 @@ def ncon(
                             out_order=out_order, check_network=check_network,
                             backend=backend)
   output = Tensor(res, backend=backend)
-  return output
+  return output
```
Comment (mganahl) [inline tensornetwork/linalg/operations.py:346]: what kind of weird character is there are the end of this line?

File: tensornetwork/backends/pytorch/pytorch_backend.py [hunk 2]
```diff
@@ -359,5 +359,4 @@ def sum(self,
   def matmul(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
     if (tensor1.ndim <= 1) or (tensor2.ndim <= 1):
       raise ValueError("inputs to `matmul` have to be a tensors of order > 1,")
-
-    return torchlib.einsum('mab,mbc->mac', tensor1, tensor2)
+    return torchlib.einsum('...ab,...bc->...ac', tensor1, tensor2)
```
Comment (mganahl) [inline tensornetwork/backends/pytorch/pytorch_backend.py:362]: good catch

File: tensornetwork/linalg/operations.py (hunk 1 above)
Comment (alewis) [reply tensornetwork/linalg/operations.py:346]: I don't see a weird character

Comment (mganahl) [reply tensornetwork/linalg/operations.py:346]: Screen Shot 2020-07-15 at 8 54 25 PM

**Pull request merged**